src/Orleans/Serialization/TypeUtilities.cs

```diff
@@ -80,6 +80,11 @@ internal static bool IsOrleansShallowCopyable(this Type t)
             return false;
         }
 
+        private static bool IsTypeFieldsShallowCopyable(TypeInfo typeInfo)
+        {
+            return typeInfo.GetFields().All(f => !(f.FieldType == typeInfo) && IsOrleansShallowCopyable(f.FieldType));
```

Comment:
I think it is a wrong comparison now.
t used to be Type, now you compare f.FieldType to TypeInfo.